Sources/Diagnose/DiagnoseCommand.swift

```diff
@@ -101,26 +108,130 @@ public struct DiagnoseCommand: AsyncParsableCommand {
+    }
+  }
+
+  private func addOsLog(toBundle bundlePath: URL) async throws {
+    #if os(macOS)
+    print("-- Collecting log messages")
+    let outputFileUrl = bundlePath.appendingPathComponent("log.txt")
+    FileManager.default.createFile(atPath: outputFileUrl.path, contents: nil)
+    let fileHandle = try FileHandle(forWritingTo: outputFileUrl)
+    let process = Process(
+      arguments: [
+        "/usr/bin/log",
+        "show",
+        "--predicate", #"subsystem = "org.swift.sourcekit-lsp" AND process = "sourcekit-lsp""#,
+        "--info",
+        "--debug",
+      ],
+      outputRedirection: .stream(
+        stdout: { try? fileHandle.write(contentsOf: $0) },
+        stderr: { _ in }
+      )
+    )
+    try process.launch()
+    try await process.waitUntilExit()
+    #endif
+  }
+
+  private func addCrashLogs(toBundle bundlePath: URL) throws {
+    #if os(macOS)
+    print("-- Collecting crash reports")
+
+    let destinationDir = bundlePath.appendingPathComponent("crashes")
+    try FileManager.default.createDirectory(at: destinationDir, withIntermediateDirectories: true)
+
+    let diagnosticReports = URL(filePath: "/Library/Logs/DiagnosticReports")
+    let enumerator = FileManager.default.enumerator(at: diagnosticReports, includingPropertiesForKeys: nil)
+    while let fileUrl = enumerator?.nextObject() as? URL {
+      guard
+        fileUrl.lastPathComponent.hasPrefix("SourceKitService") || fileUrl.lastPathComponent.hasPrefix("sourcekit-lsp")
+      else {
+        continue
+      }
+      try? FileManager.default.copyItem(
+        at: fileUrl,
+        to: destinationDir.appendingPathComponent(fileUrl.lastPathComponent)
+      )
+    }
+    #endif
+  }
+
+  private func addSwiftVersion(toBundle bundlePath: URL) async throws {
+    print("-- Collecting installed Swift versions")
+
+    let outputFileUrl = bundlePath.appendingPathComponent("swift-versions.txt")
+    FileManager.default.createFile(atPath: outputFileUrl.path, contents: nil)
+    let fileHandle = try FileHandle(forWritingTo: outputFileUrl)
 
-    guard let reproducerBundle else {
-      print("No reducible crashes found")
-      throw ExitCode(1)
+    for toolchain in try await toolchainRegistry.toolchains {
+      guard let swiftUrl = toolchain.swift?.asURL else {
+        continue
+      }
+
+      try fileHandle.write(contentsOf: "\(swiftUrl.path) --version\n".data(using: .utf8)!)
+      let process = Process(
+        arguments: [swiftUrl.path, "--version"],
+        outputRedirection: .stream(
+          stdout: { try? fileHandle.write(contentsOf: $0) },
+          stderr: { _ in }
+        )
+      )
+      try process.launch()
+      try await process.waitUntilExit()
+      fileHandle.write("\n".data(using: .utf8)!)
     }
+  }
+
+  public func run() async throws {
+    print(
+      """
+      sourcekit-lsp diagnose collects information that helps developers of sourcekit-lsp diagnose and fix issues. 
+      This information contains:
```

Comment:
Join to previous and make `This includes:`?

Reply:
Which previous do you mean?